features/topology-map/org.opennms.features.topology.app/src/main/java/org/opennms/features/topology/app/internal/TopologyUI.java

```diff
@@ -206,11 +208,26 @@ private TopologyUIRequestHandler() {
                     // The order matters
                     this::loadHistoryFragment,
                     this::loadGraphProvider,
+                    this::loadLayer,
                     this::loadVertexHopCriteria,
                     this::loadSemanticZoomLevel,
                     this::loadLayout);
         }
 
+        private boolean loadLayer(VaadinRequest request) {
```

Comment:
nit pick: there are quite a few conditions within conditions in this code block. But I guess there is no nicer easy way to express this....